src/doip_simulator.cpp

```diff
@@ -0,0 +1,188 @@
+#include "doip_simulator.h"
+
+/**
+ * Constructor. Creates a DoIPServer for this simulator
+ */
+DoIPSimulator::DoIPSimulator() {
+    doipserver = new DoIPServer();
+}
+
+/**
+ * Initialize server instance with required callbacks, start the doip server
+ */
+void DoIPSimulator::start() {
+    DiagnosticCallback cb = std::bind(&DoIPSimulator::receiveFromLib, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
```

Comment:
Variable name "cb" is not  self-descriptive

Reply:
Thank you, changed this and some other variable names to hopefully more self-descriptive names.